src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
>Shouldn't we test that server and client are in the same sub-net?

I don't think we need this check, different clients in different subnets won't be able to talk to each other directly, well... unless someone explicitly set it up...

>For instance, your guesthouse gives IPs from sub-net 10.0.0.0/24

In this example, I would say if someone on 10.0.0.0/24 is able to connect to 192.168.0.0/16 freely, then that router has serious security flaw (routing table or firewall had been tampered), I don't think is the common case.